src/Forms/MoneyField.php

```diff
@@ -321,32 +338,30 @@ public function getLocale()
         return $this->fieldAmount->getLocale();
     }
 
-    /**
-     * Validate this field
-     *
-     * @param Validator $validator
-     * @return bool
-     */
-    public function validate($validator)
+    public function getValueForValidation(): mixed
     {
-        // Validate currency
-        $currencies = $this->getAllowedCurrencies();
-        $currency = $this->fieldCurrency->dataValue();
-        if ($currency && $currencies && !in_array($currency, $currencies ?? [])) {
-            $validator->validationError(
-                $this->getName(),
-                _t(
-                    __CLASS__ . '.INVALID_CURRENCY',
-                    'Currency {currency} is not in the list of allowed currencies',
-                    ['currency' => $currency]
-                )
-            );
-            return $this->extendValidationResult(false, $validator);
-        }
+        return [$this->getAmountField(), $this->getCurrencyField()];
+    }
 
-        // Field-specific validation
-        $result = $this->fieldAmount->validate($validator) && $this->fieldCurrency->validate($validator);
-        return $this->extendValidationResult($result, $validator);
+    public function validate(): ValidationResult
+    {
+        $result = ValidationResult::create();
+        $this->beforeExtending('updateValidate', function () use ($result) {
```

Comment:
```suggestion
        $this->beforeExtending('updateValidate', function (ValidationResult $result) {
```
I've asked this in a previous review: Please double check all of the changed fields to ensure you are doing it this way instead, so the same result passed into the extension hook is used.
This is the only one I found this time around but I may have missed one.

Reply:
I think this was the only one to slip through, have updated